File: backend/src/main/java/outcobra/server/model/SchoolYear.java
package outcobra.server.model;

import java.time.LocalDate;
import java.util.List;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class SchoolYear {
    @Id
    private Long id;

    @NotNull
    private String name;

    @NotNull
    private LocalDate validFrom, validTo;

    @ManyToOne
    private SchoolClass schoolClass;

    @OneToMany(mappedBy = "schoolYear")
    private List<Holiday> holidays;

    @OneToMany(mappedBy = "schoolYear")
    private List<Semester> semesters;
}

File: backend/src/main/kotlin/outcobra/server/model/SchoolYear.kt
package outcobra.server.model

import com.querydsl.core.annotations.QueryInit
import outcobra.server.model.interfaces.ParentLinked
import java.time.LocalDate
import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.validation.constraints.NotNull

@Entity
class SchoolYear(
        @NotNull var name: String = "",
        @NotNull var validFrom: LocalDate = LocalDate.now(),
        @NotNull var validTo: LocalDate = LocalDate.now(),
        @QueryInit("institution.user") @ManyToOne var schoolClass: SchoolClass? = null,
        @OneToMany(mappedBy = "schoolYear") var holidays: List<Holiday> = listOf(),
        @OneToMany(mappedBy = "schoolYear", cascade = arrayOf(CascadeType.REMOVE))
        var semesters: List<Semester> = listOf()) : ParentLinked, AbstractEntity() {
    override val parent: ParentLinked?
        get() = schoolClass
}